### PR TITLE
Enable long polling strategy by default

### PR DESCRIPTION
**Pull request by randomir**

### CODE REVIEW

**Review by arcondello — APPROVED** [review]

File: dwave/cloud/client/base.py [hunk 1]
```diff
@@ -173,24 +173,49 @@ class Client(object):
         client_cert_key (str, optional):
             Path to client side certificate key file.
 
+        poll_strategy (str, 'backoff' | 'long-polling', default='long-polling'):
+            Problem status polling strategy. Chose between short-polling with
+            exponential back-off (configured with `poll_backoff_*` parameters),
+            and long polling, configured with `poll_wait_time` and `poll_pause`
+            parameters.
```
Comment (JoelPasvolsky) [inline dwave/cloud/client/base.py:180]:
```suggestion
            Polling strategy for problem status. Supported options are short polling 
            with exponential back-off, configured with ``poll_backoff_*`` parameters,
            and long polling, configured with ``poll_wait_time`` and ``poll_pause``
            parameters.
```

File: dwave/cloud/client/base.py [hunk 2]
```diff
@@ -173,24 +173,49 @@ class Client(object):
...
+
+            .. versionadded:: 0.13.0
+                Added long polling strategy support. Long polling is the new
```
Comment (JoelPasvolsky) [inline dwave/cloud/client/base.py:183]:
```suggestion
                Added support for long polling strategy. Long polling is the new
```

File: dwave/cloud/client/base.py [hunk 3]
```diff
@@ -173,24 +173,49 @@ class Client(object):
...
             Interval between ``poll_idx`` and ``poll_idx + 1`` is given with::
 
                 poll_backoff_min * poll_backoff_base ** poll_idx
 
             with upper bound set to ``poll_backoff_max``.
 
+        poll_wait_time (int, in range [0, 30], default=30):
+            When problem status is polled using long polling, the maximum time a
+            long polling connection will wait for the API response is given with
+            ``poll_wait_time`` in seconds.
```
Comment (JoelPasvolsky) [inline dwave/cloud/client/base.py:208]:
```suggestion
            When problem status is polled using long polling, this value sets the 
            maximum time, in seconds, a long polling connection waits for the API 
            response.
```

File: dwave/cloud/client/base.py [hunk 4]
```diff
@@ -173,24 +173,49 @@ class Client(object):
...
+
+            .. versionadded:: 0.13.0
+
+        poll_pause (float, default=0.0):
+            When problem status is polled using long polling, a delay introduced
+            between two successive long polling connections is limited to
+            ``poll_pause`` seconds. It's preferred to keep this value at zero.
```
Comment (JoelPasvolsky) [inline dwave/cloud/client/base.py:215]:
```suggestion
            When problem status is polled using long polling, this value limits the 
            delay, in seconds, between two successive long polling connections.
            It's preferred to keep this value at zero.
```

File: releasenotes/notes/default-to-long-polling-a3bb3254a42c794b.yaml [hunk 5]
```diff
@@ -0,0 +1,17 @@
+---
+features:
+  - |
+    Poll for problem status update using long polling strategy by default.
```
Comment (JoelPasvolsky) [inline releasenotes/notes/default-to-long-polling-a3bb3254a42c794b.yaml:4]:
```suggestion
    Poll for problem-status update using long-polling strategy by default.
```

File: releasenotes/notes/default-to-long-polling-a3bb3254a42c794b.yaml [hunk 6]
```diff
@@ -0,0 +1,17 @@
...
+
+    Long polling strategy enables faster status change propagation to the client,
```
Comment (JoelPasvolsky) [inline releasenotes/notes/default-to-long-polling-a3bb3254a42c794b.yaml:6]:
```suggestion
    Long-polling strategy enables faster status-change propagation to the client,
```

**Review by JoelPasvolsky — APPROVED** [review]
LGTM, thanks!

**Pull request merged**